packages/apollo-server-memcached/src/index.ts

```diff
@@ -2,31 +2,109 @@ import { KeyValueCache } from 'apollo-server-caching';
 import Memcached from 'memcached';
 import { promisify } from 'util';
 
+/**
+ * Defines a KeyValueCache implementation that supports cache tag invalidation.
+ * When cache entry is associated with 1 or more tags, we store tag:version
+ * pairs as metadata, as well as the current version associated with each tag.
+ * Version numbers are simply integers that are incremented whenever a tag is
+ * invalidated. For a cache entry to be valid, all of its associated tags must
+ * have version numbers that match their current version numbers.
+ *
+ * Performance:
+ * - set()        : # of db reads = # of tags
+ * - get()        : # of db reads = # of tags
+ * - invalidate() : # of db writes/incr = 1 for each tag
+ *
+ * Storage overhead:
+ * - key->data + tag:version pairs for each tag
+ * - tag->version stored for each tag
+ */
 export default class MemcachedKeyValueCache implements KeyValueCache {
   readonly client;
```

Comment:
I don't think `client` should be `any`, but we want a 'promisified' version of the `Memcached` interface.

Reply:
I'm not sure if there is a nice way to get the promisified Memcached type easily, since most libraries that I found lose the type information on promisification. The best option now would be to redeclare a different type manually, with only the methods calls that we need?